campaignion_donation/components/donation_amount.php

```diff
@@ -0,0 +1,213 @@
+    '#attributes' => ['class' => ['other-option-checkbox']],
+  ];
+  $form['behavior']['other_text'] = [
+    '#type' => 'textfield',
+    '#title' => t('Text for "Other..." option'),
+    '#default_value' => $component['extra']['other_text'],
+    '#description' => t('If allowing other options, enter text to be used for other-enabling option.'),
+    '#parents' => ['extra', 'other_text'],
+    '#weight' => 3,
+    '#states' => [
+      'visible' => [
+        ':input.other-option-checkbox' => ['checked' => TRUE],
+      ],
+    ],
+  ];
+
+  return $form;
+}
+
+/**
+ * Element validate callback: Split amount array.
+ */
+function _webform_edit_donation_amount_validate_amounts($element, &$form_state) {
+  $values =
+
+  $validate_number = function ($value) use ($element, $form_state) {
+    $component = $form_state['values'];
+    $element_stub = [
+      '#title' => $element['#title'],
+      '#min' => $component['extra']['min'],
+      '#max' => $component['extra']['max'],
+      '#step' => $component['extra']['step'] ? abs($component['extra']['step']) : '',
+      '#integer' => $component['extra']['integer'],
+      '#point' => '.',
+      '#separator' => ',',
+      '#decimals' => $component['extra']['decimals'],
+      '#parents' => ['number'],
+      '#value' => $value,
+    ];
+    $fs['values'] = [];
+
+    // Backup $form_state and $_SESSION['messages'] and restore them afterwards.
+    $messages = $_SESSION['messages'] ?? [];
+    $errors = &drupal_static('form_set_error', []);
+    $original_errors = $errors;
+    $errors = [];
+    _webform_validate_number($element_stub, $fs);
+    $our_errors = $errors;
+    $errors = $original_errors;
+    $_SESSION['messages'] = $messages;
+
+    if ($our_errors) {
+      return FALSE;
+    }
+    return webform_number_standardize($value, $element_stub['#point']);
+  };
+
+  $only_numeric = TRUE;
+  $values = [];
+  foreach (array_filter(array_map('trim', explode("\n", $element['#value']))) as $value) {
+    if (($number = $validate_number($value)) !== FALSE) {
+      $values[] = $number;
+    }
+    else {
+      $only_numeric = FALSE;
+      break;
+    }
+  }
+  if (!$only_numeric) {
+    form_error($element, t('Only valid numeric values are allowed (see validation settings).'));
+    return;
+  }
+  form_set_value($element, $values, $form_state);
+}
+
+/**
+ * Implements _webform_render_component().
+ */
+function _webform_render_donation_amount($component, $value = NULL, $filter = TRUE, $submission = NULL) {
+  $number_component = $component;
+  $component['extra']['type'] = 'textfield';
+  $component['extra']['unique'] = FALSE;
+  $element = webform_component_invoke('number', 'render', $component, $value, $filter, $submission);
+  $element['#attributes']['class'][] = 'donation-amount';
+  if ($component['extra']['options']) {
+    $element['#type'] = 'select_or_other';
+    $element['#select_type'] = 'radios';
+    $element['#options'] = drupal_map_assoc($component['extra']['options']);
```

Comment:
Would be great if we can toggle whether the currency symbol gets prepended before the amount. This is a very common use case in rendering. (Just rendering it an hiding via styles would work too)

The currency symbol would be best wrapped in a span with an appropriate class...